### ISSUE TITLE
Warning when navigating to site: "Warning: A props object containing a "key" prop is being spread into JSX"

### ISSUE THREAD

**Issue by tm-ruxandra** [techmatters/terraso-mobile-client#2223]
Steps to repro:
1) Open application in developer mode
2) Navigate to any site
3) Observe the following error onscreen:

```
 ERROR  Warning: A props object containing a "key" prop is being spread into JSX:
  let props = {key: someKey, position: ..., route: ..., navigationState: ..., getAccessibilityLabel: ..., getAccessible: ..., getLabelText: ..., getTestID: ..., renderBadge: ..., renderIcon: ..., renderLabel: ..., activeColor: ..., inactiveColor: ..., pressColor: ..., pressOpacity: ..., onLayout: ..., onPress: ..., onLongPress: ..., labelStyle: ..., style: ..., defaultTabWidth: ..., android_ripple: ...};
  <TabBarItem {...props} />
React keys must be passed directly to JSX without using spread:
  let props = {position: ..., route: ..., navigationState: ..., getAccessibilityLabel: ..., getAccessible: ..., getLabelText: ..., getTestID: ..., renderBadge: ..., renderIcon: ..., renderLabel: ..., activeColor: ..., inactiveColor: ..., pressColor: ..., pressOpacity: ..., onLayout: ..., onPress: ..., onLongPress: ..., labelStyle: ..., style: ..., defaultTabWidth: ..., android_ripple: ...};
```

`TabBarItem` seems to be somewhere upstream of our navigation system; is it possible that some updated dependency made our usage invalid?

<img width="323" alt="Screenshot 2024-10-01 at 2 39 48 PM" src="https://github.com/user-attachments/assets/0e8c8566-86dc-46d1-90db-db4f8462347b">

**Comment by knipec** [comment]
Triage notes: 
- It's a bug in the react navigation library. It's been fixed but not released: https://github.com/react-navigation/react-navigation/pull/12142

**Comment by semihyagli** [comment]
When will this be released? I see that the latest release was on July 25 and this PR was just merged last week.

**Comment by tm-ruxandra** [comment]
@semihyagli We are not the maintainers of the react navigation library, this is just a project which was affected by the issue. Please direct your questions regarding the issue to their repository.

**Comment by paulschreiber** [comment]
This was fixed in react-native-tab-view 4.0.0.

**Comment by DerekCaelin** [comment]
@tm-ruxandra could you verify this, it looks like you need to be in developer mode to review?
